charts/newrelic-logging/templates/daemonset.yaml

```diff
@@ -47,8 +47,17 @@ spec:
       securityContext:
         {{- . | nindent 8 }}
       {{- end }}
-      {{- if .Values.initContainers }}
       initContainers:
+      {{- if and (.Values.fluentBit.persistence) (eq .Values.fluentBit.persistence.mode "persistentVolume") }}
```

Comment:
General comment (for here and the rest of the PR). If we set a default value for `.Values.fluentBit.persistence.mode` in the `values.yml` file, I suspect we don't need to do `and (.Values.fluentBit.persistence) (eq .Values.fluentBit.persistence.mode "<something>")` everywhere, right? We can just go for `{{- if eq .Values.fluentBit.persistence.mode "<something>" }}`

Reply:
We need it for upgrades or that's demonstrated by the tests included. If it tries to upgrade using previous values it'll fail. https://github.com/newrelic/helm-charts/actions/runs/7669042487/job/20902151151

Reply:
I see, thanks for verifying